### ISSUE TITLE
[Regression] jsdoc/no-undefined-types warning for module:stream.Readable

### ISSUE THREAD

**Issue by kevinoid** [gajus/eslint-plugin-jsdoc#747]
## Expected behavior

`jsdoc/no-undefined-types` does not warn for `module:stream.Readable`.

## Actual behavior

    index.js:4:0: The type 'Readable' is undefined. [Warning/jsdoc/no-undefined-types]

## ESLint Config

```json
{
  "env": {
    "es6": true,
    "node": true
  },
  "extends": [
    "plugin:jsdoc/recommended"
  ]
}
```

## ESLint sample

```js
/**
 * Read all content from a Readable stream.
 *
 * @param {!module:stream.Readable} readable Readable stream.
 * @returns {!Promise<!Array<*>>} Content from readable.
 */
module.exports =
function readStream(readable) {
  return new Promise((resolve, reject) => {
    const data = [];
    readable
      .on('data', (d) => data.push(d))
      .once('error', reject)
      .once('end', () => resolve(data));
  });
};
```

## Environment

- Node version: v12.21.0
- ESLint version v7.27.0
- `eslint-plugin-jsdoc` version: v35.1.1 or v35.1.0

## Additional Info

The issue does not occur with `eslint-plugin-jsdoc@35.0.0` or earlier.  I bisected the regression to e9f6a0a from #736.

I'm still not entirely clear on what types JSDoc accepts.  My apologies in advance if `module:stream.Readable` is not a valid type.

Thanks,
Kevin

**Comment by simonseyock** [comment]
Thanks for reporting this! This should be fixed in https://github.com/simonseyock/jsdoc-type-pratt-parser/pull/60.

**Comment by kevinoid** [comment]
Thanks for the fast response and fix @simonseyock!  Much appreciated.  I can confirm that `jsdoc-type-pratt-parser@1.0.2` fixes the issue for me.

I noticed a couple other potential issues.  Would you prefer if I open them in this repo or [simonseyock/jsdoc-type-pratt-parser](https://github.com/simonseyock/jsdoc-type-pratt-parser)?  (Or as comments on this issue?)

**Comment by simonseyock** [comment]
Feel free to report them on simonseyock/jsdoc-type-pratt-parser

**Comment by brettz9** [comment]
@kevinoid , if these are not very obscure issues (like the non-allowance of underscores as was just fixed in the parser), feel free to also file here so others can see here that it is being worked on and so that we can track our updating of the parser once it may include a fix/enhancement.
